File: 0661-image-smoother/0661-image-smoother.py
class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        directions = ((0, 0), (-1, 1), (0, 1), (1, 1), (1, 0), (1, -1), (0, -1), (-1, -1), (-1, 0))
        m, n = len(img), len(img[0])
        filtered = [[0] * n for _ in range(m)]
        
        for y in range(m):
            for x in range(n):
                total, cells = 0, 0
                for dx, dy in directions:
                    if 0 <= y + dy < m and 0 <= x + dx < n:
                        total += img[y + dy][x + dx]
                        cells += 1
                filtered[y][x] = total // cells
        return filtered

File: 0661-image-smoother/0661-image-smoother.cpp
class Solution {
public:
    vector<vector<int>> imageSmoother(vector<vector<int>>& img) {
        vector<pair<int, int>> directions{{0, 0}, {-1, 1}, {0, 1},\
        {1, 1}, {1, 0}, {1, -1},\
        {0, -1}, {-1, -1}, {-1, 0}
                                         };
        int m = img.size(), n = img[0].size();
        vector filtered(m, vector<int>(n));
        
        for (int y = 0; y < m; ++y){
            for (int x = 0; x < n; ++x){
                int total = 0, cells = 0;
                for (auto& [dy, dx] : directions){
                    if (0 <= y + dy && y + dy < m && 0 <= x + dx && x + dx < n){
                        total += img[y + dy][x + dx];
                        ++cells;
                    }
                }
                filtered[y][x] = total / cells;
            }
        }
        
        return filtered;
    }
};